Core_OverlayMods/Clothes/KoiClothesOverlayController.cs

```diff
@@ -118,6 +138,128 @@ public static bool IsMaskKind(string clothesId)
             return false;
 #endif
         }
+
+        public static string MakeColormaskId(string clothesId)
+        {
+
+            return $"Colormask_{clothesId}";
+        }
+
+        public static bool IsColormask(string clothesId)
+        {
+            return clothesId.StartsWith("Colormask_");
+        }
+
+        public static bool GetKindIdsFromColormask(string clothesId, out int? kindId, out int? subKindId)
+        {
+            kindId = null;
+            subKindId = null;
+
+            if (IsColormask(clothesId))
+            {
+                switch (GetRealId(clothesId))
+                {
+                    case "ct_top_parts_A":
+                        kindId = 0;
+                        subKindId = 0;
+                        break;
+                    case "ct_top_parts_B":
+                        kindId = 0;
+                        subKindId = 1;
+                        break;
+                    case "ct_top_parts_C":
+                        kindId = 0;
+                        subKindId = 2;
+                        break;
+                    case "ct_clothesTop":
+                        kindId = 0;
+                        break;
+                    case "ct_clothesBot":
+                        kindId = 1;
+                        break;
+#if KK || KKS || EC
+                    case "ct_bra":
+                        kindId = 2;
+                        break;
+                    case "ct_shorts":
+                        kindId = 3;
+                        break;
+#else
+                    case "ct_inner_t":
+                        kindId = 2;
+                        break;
+                    case "ct_inner_b":
+                        kindId = 3;
+                        break;
+#endif
+                    case "ct_gloves":
+                        kindId = 4;
+                        break;
+                    case "ct_panst":
+                        kindId = 5;
+                        break;
+                    case "ct_socks":
+                        kindId = 6;
+                        break;
+#if KK || KKS
+                    case "ct_shoes_inner":
+                        kindId = 7;
+                        break;
+                    case "ct_shoes_outer":
+                        kindId = 8;
+                        break;
+#else
+                    case "ct_shoes":
+                        kindId = 7;
+                        break;
+#endif
+                }
+                return true;
+            }
+            return false;
+        }
+
+        public static string GetRealId(string clothesId)
+        {
+            if (IsColormask(clothesId))
+                return clothesId.Substring(10);
```

Comment:
`Colormask_` should be made into a const and its Length used here instead of 10.